pynest/examples/Potjans_2014/network.py

```diff
@@ -0,0 +1,477 @@
+# -*- coding: utf-8 -*-
+#
+# network.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+'''
+pynest microcicuit network
+--------------------------
+
+Main file for the microcircuit.
+
+Hendrik Rothe, Hannah Bos, Sacha van Albada; May 2016
+'''
+
+import nest
+import numpy as np
+import os
+from helpers import adj_w_ext_to_K
+from helpers import synapses_th_matrix
+from helpers import get_total_number_of_synapses
+from helpers import get_weight
+from helpers import plot_raster
+from helpers import fire_rate
+from helpers import boxplot
+from helpers import compute_DC
+
+
+class Network:
+    """ Handles the setup of the network parameters and
+    provides functions to connect the network and devices.
+
+    Arguments
+    ---------
+    sim_dict
+        dictionary containing all parameters specific to the simulation
+        such as the directory the data is stored in and the seeds
+        (see: sim_params.py)
+    net_dict
+         dictionary containing all parameters specific to the neurons
+         and the network (see: network_params.py)
+
+    Keyword Arguments
+    -----------------
+    stim_dict
+        dictionary containing all parameter specific to the stimulus
+        (see: stimulus_params.py)
+
+    """
+    def __init__(self, sim_dict, net_dict, stim_dict=None):
+        self.sim_dict = sim_dict
+        self.net_dict = net_dict
+        if stim_dict is not None:
+            self.stim_dict = stim_dict
+        else:
+            self.stim_dict = None
+        self.data_path = sim_dict['data_path']
+        if nest.Rank() == 0:
+            if os.path.isdir(self.sim_dict['data_path']):
+                print('data directory already exists')
+            else:
+                os.mkdir(self.sim_dict['data_path'])
+                print('data directory created')
+        print('Data will be written to %s' % self.data_path)
```

Comment:
If used with MPI such statements are currently printed once per process. Please print them only on Rank 0 unless you want to print something specific to a process.